File: Api/src/models/ship.ts
import mongoose, {Document, Schema} from 'mongoose';

export interface IShip extends Document {
  shipId: number,
  emailUsername: string,
  name: string,
  img:
  {
    data: Buffer,
    contentType: String
  },
}

const ShipSchema: Schema = new Schema({
  shipId: Number,
  emailUsername: String,
  name: String,
  img:
  {
    data: Buffer,
    contentType: String,
    select: false,
  },
});

export default mongoose.model<IShip>('Ship', ShipSchema);

File: App/App/Models/Ship.cs
using System;

namespace Oxbridge.App.Models
{
    public class Ship
    {
        public int ShipId { get; set; }
        public String Name { get; set; }
        public String EmailUsername { get; set; }
        public String TeamName { get; set; }
        public Image Img { get; set; }
    }
}
